skore-ui/src/components/CacheableComponent.vue

```diff
@@ -0,0 +1,77 @@
+<script setup lang="ts">
```

Comment:
In some way "cacheable" means that the component uses resources to build or use, and in order to spare those resources, we cache it.

So I was thinking a better name could be "DynamicContentRasterizer".
And here is the source code for it:

```vue
<script setup lang="ts">
import { debounce } from "@/services/utils";
import { toPng } from "html-to-image";
import { computed, onBeforeUnmount, ref, useTemplateRef, watch } from "vue";

const props = defineProps<{
  isRasterized: boolean;
}>();

const rasterizedSrc = ref("");
const realContent = useTemplateRef("realContent");
const contentHeight = ref(0);

const styles = computed(() => ({
  height: contentHeight.value !== 0 ? `${contentHeight.value}px` : "auto",
}));

const updateRasterizedImage = debounce(
  async () => {
    if (realContent.value) {
      rasterizedSrc.value = await toPng(realContent.value as HTMLElement);
    }
  },
  100,
  false
);

const resizeObserver = new ResizeObserver(async (entries) => {
  if (entries.length == 1 && !props.isRasterized) {
    const content = entries[0];
    contentHeight.value = content.contentRect.height;
    updateRasterizedImage();
  }
});

watch(
  () => realContent.value,
  () => {
    if (realContent.value) {
      resizeObserver.observe(realContent.value as HTMLElement);
    }
  }
);

onBeforeUnmount(() => {
  resizeObserver.disconnect();
});
</script>

<template>
  <div class="dynamic-content-rasterizer" :style="styles">
    <div class="real-content" v-if="!isRasterized" ref="realContent">
      <slot />
    </div>
    <div class="rasterized-content" v-else>
      <img :src="rasterizedSrc" alt="Rasterized content" />
    </div>
  </div>
</template>

<style scoped>
.dynamic-content-rasterizer {
  position: relative;
  padding: 0;
  margin: 0;
}

.rasterized-content {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
}
</style>
```

Reply:
Rename done : )